xbmc/dbwrappers/sqlitedataset.cpp

```diff
@@ -21,6 +23,112 @@
 #include "platform/linux/XTimeUtils.h"
 #endif
 
+namespace {
+#define X(VAL) std::make_pair(VAL, #VAL)
+const std::map<int, const char*> g_SqliteErrorStrings =
+{
+  X(SQLITE_OK),
+  X(SQLITE_ERROR),
+  X(SQLITE_INTERNAL),
+  X(SQLITE_PERM),
+  X(SQLITE_ABORT),
+  X(SQLITE_BUSY),
+  X(SQLITE_LOCKED),
+  X(SQLITE_NOMEM),
+  X(SQLITE_READONLY),
+  X(SQLITE_INTERRUPT),
+  X(SQLITE_IOERR),
+  X(SQLITE_CORRUPT),
+  X(SQLITE_NOTFOUND),
+  X(SQLITE_FULL),
+  X(SQLITE_CANTOPEN),
+  X(SQLITE_PROTOCOL),
+  X(SQLITE_EMPTY),
+  X(SQLITE_SCHEMA),
+  X(SQLITE_TOOBIG),
+  X(SQLITE_CONSTRAINT),
+  X(SQLITE_MISMATCH),
+  X(SQLITE_MISUSE),
+  X(SQLITE_NOLFS),
+  X(SQLITE_AUTH),
+  X(SQLITE_FORMAT),
+  X(SQLITE_RANGE),
+  X(SQLITE_NOTADB),
+  X(SQLITE_NOTICE),
+  X(SQLITE_WARNING),
+  X(SQLITE_ROW),
+  X(SQLITE_DONE),
+  X(SQLITE_ERROR_MISSING_COLLSEQ),
```

Comment:
> error C2065: 'SQLITE_ERROR_MISSING_COLLSEQ': undeclared identifier

some more errors about undeclared identifier on windows

Reply:
What's the windows sqlite version?

Reply:
lowest one is 3.10.2

Reply:
Should they be updated?

Reply:
+1 for update

Reply:
3.10.2 is 3 years old

Reply:
Update for v19 sure, but to do so in point release has some risks

Reply:
Ubuntu Xenial (16.04LTS) has 3.11.0, so same errors will happen there.

Reply:
ok